File: MenuPopOverView/ViewController.m
//
//  ViewController.m
//  MenuPopOverView
//
//  Created by Camel Yang on 4/4/14.
//  Copyright (c) 2014 camelcc. All rights reserved.
//

#import "ViewController.h"

#import "MenuPopOverView.h"

@interface ViewController () <MenuPopOverViewDelegate>

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapped:)];
    [self.view addGestureRecognizer:tap];
}

- (void)tapped:(UITapGestureRecognizer *)tap
{
    CGPoint point = [tap locationInView:self.view];
    MenuPopOverView *popOver = [[MenuPopOverView alloc] init];
    popOver.delegate = self;
    [popOver presentPopoverFromRect:CGRectMake(point.x, point.y, 0, 0) inView:self.view withStrings:@[@"Test1", @"TestAAAAAAA", @"t", @"example", @"loooooooooooooooongbutton"]];
}

- (void)popoverView:(MenuPopOverView *)popoverView didSelectItemAtIndex:(NSInteger)index {
    NSLog(@"select at %ld", (long)index);
}

- (void)popoverViewDidDismiss:(MenuPopOverView *)popoverView {
    NSLog(@"popOver dismissed.");
}

@end

File: Demo/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by camel_yang on 1/30/18.
//  Copyright © 2018 camelcc. All rights reserved.
//

import UIKit
import MenuPopOverView

class ViewController: UIViewController {
    var menuPopOverView: MenuPopOverView? = nil

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

        self.view.addGestureRecognizer(UITapGestureRecognizer.init(target: self, action: #selector(tapped(tap:))))
    }

    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
        super.viewWillTransition(to: size, with: coordinator)
        menuPopOverView?.dismiss(false)
    }

    @objc private func tapped(tap: UITapGestureRecognizer) {
        let tapLocation = tap.location(in: self.view)
        if menuPopOverView == nil {
            menuPopOverView = MenuPopOverView()
        }
        menuPopOverView?.delegate = self
        menuPopOverView?.present(at: CGRect(x: tapLocation.x, y: tapLocation.y, width: 0, height: 0),
                                in: self.view,
                                with: ["Test1", "TestAAAAAAA", "t", "example", "loooooooooooooooongbutton"])
    }
}

extension ViewController: MenuPopOverViewDelegate {
    func didSelect(view: MenuPopOverView, at index: Int) {
        print("didSelect at \(index)")
    }

    func didDismiss(view: MenuPopOverView) {
        print("didDismiss")
        menuPopOverView = nil
    }
}

